dataprofiler/labelers/base_model.py

```diff
@@ -135,17 +156,51 @@ def set_params(self, **kwargs):
         for param in kwargs:
             self._parameters[param] = kwargs[param]
 
+    def add_label(self, label, same_as=None):
+        """
+        Adds a label to the data labeler.
+
+        :param label: new label being added to the data labeler
+        :type label: str
+        :param same_as: label to have the same encoding index as for multi-label
+            to single encoding index.
+        :type same_as: str
+        :return: None
+        """
+        # validate label
+        if not label or not isinstance(label, str):
+            raise TypeError('`label` must be a str.')
+        elif label in self._label_mapping:
+            warnings.warn('The label, `{}`, already exists in the label '
+                          'mapping.'.format(label))
+            return
+
+        # validate same_as
+        if same_as and not isinstance(same_as, str):
+            raise TypeError('`same_as` must be a str.')
+        elif same_as and same_as not in self._label_mapping:
+            raise ValueError('`same_as` value: {}, did not exist in the '
+                             'label_mapping.'.format(same_as))
+
+        # add label to label_mapping
+        max_label_ind = max(self._label_mapping.values())
+        self._label_mapping[label] = self._label_mapping.get(same_as,
+                                                             max_label_ind + 1)
+
     def set_label_mapping(self, label_mapping):
         """
         Sets the labels for the model
 
-        :param label_mapping: label mapping of the model
-        :type label_mapping: dict
+        :param label_mapping: label mapping of the model or list of labels to be
+            converted into the label mapping
+        :type label_mapping: Union[list, dict]
         :return: None
         """
-        if not label_mapping or not isinstance(label_mapping, dict):
-            raise ValueError("`label_mapping` must be a dict which maps labels "
-                             "to index encodings.")
+        if not isinstance(label_mapping, (list, dict)) or not label_mapping:
```

Comment:
Is there a point to the second half of this if statement? if label_mapping is None then it will already make the isinstance() call fail, which will short circuit and the second half of the if statement would never get reached.